src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/UserRepositoryTests.cs

```diff
@@ -0,0 +1,86 @@
+﻿#nullable enable
+using System;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Xunit;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Tests.Services.DbUtils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Tests.Services
+{
+    public class UserRepositoryTests
+    {
+        [Fact]
+        public async Task FindByEmail()
+        {
+            var user = new User
+            {
+                Email = "test@test.com"
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.Users.AddAsync(user);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var repository = SetupUserReleaseRoleRepository(contentDbContext);
+                var result = await repository.FindByEmail("test@test.com");
+                Assert.NotNull(result);
+                Assert.Equal(user.Id, result!.Id);
+            }
+        }
+
+        [Fact]
+        public async Task FindByEmail_DifferentCase()
+        {
+            var user = new User
+            {
+                Email = "test@test.com"
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.Users.AddAsync(user);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var repository = SetupUserReleaseRoleRepository(contentDbContext);
+                var result = await repository.FindByEmail("TEST@TEST.COM");
+
+                // While this test could fail using the in-memory db if the comparison was accidentally case sensitive,
+                // it wouldn't fail using SqlServer which is using a case insensitive collation by default.
```

Comment:
Case insensitive collation `SQL_Latin1_General_CP1_CI_AS` is the default for SQL Server. I've checked Azure Portal which confirms this is what we have set. We would have had to set this when creating the database and I'm not sure how safe or easy it is to change it now. Setting a different collation when initialising the database schema such as this case sensitive one `SQL_Latin1_General_CP1_CS_AS` became possible in EF 5.0 using `modelBuilder.UseCollation("SQL_Latin1_General_CP1_CS_AS")`. Since EF 5.0 its also possible to set it on individual columns.